tests/kernel/mem_pool/sys_mem_pool/src/main.c

```diff
@@ -19,10 +20,29 @@
 #define BLK_SIZE_EXCLUDE_DESC (BLK_SIZE_MIN - 16)
 #define BLK_ALIGN BLK_SIZE_MIN
 
+K_THREAD_STACK_DEFINE(tstack, STACK_SIZE);
+__kernel struct k_thread tdata;
 
 K_MUTEX_DEFINE(pool_mutex);
 SYS_MEM_POOL_DEFINE(pool, &pool_mutex, BLK_SIZE_MIN, BLK_SIZE_MAX,
 		    BLK_NUM_MAX, BLK_ALIGN, .data);
+K_SEM_DEFINE(end_sema, 0, 1);
+
+/**
+ * @see k_malloc(), k_free()
+ */
+static void thread_handler(void *p1, void *p2, void *p3)
+{
+	void *ptr;
+
+	k_thread_system_pool_assign(k_current_get());
```

Comment:
This test case is to validate the sys_mem_pool APIs as defined in include/misc/mempool.h

Why are we shoving a test for k_thread_system_pool_assign() into this test case? It does not belong here.

Reply:
@andrewboie Moved this test to tests/kernel/mem_pool/mem_pool_api/ folder.